app and creds last used date docs (#20289)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/serviceprincipalsigninactivity-get.md

Get a [servicePrincipalSignInActivity](../resources/serviceprincipalsigninactivity.md) object that contains sign-in activity information for a service principal in an Azure Active Directory tenant. You can use a service principal as a client or resource. A service principal supports delegated or app-only authentication context.

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
| :------------------------------------- | :------------------------------------------ |
| Delegated (work or school account)     | AuditLog.Read.All, Directory.Read.All       |
| Delegated (personal Microsoft account) | Not supported.                              |
| Application                            | AuditLog.Read.All, Directory.Read.All       |

## HTTP request

<!-- { "blockType": "ignored" } -->
```http
GET /reports/servicePrincipalSignInActivities/{servicePrincipalSignInActivityId}
```

## Optional query parameters

> This method supports the `$top`, `$filter`, and `$orderBy` OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

Reviewer: Same comment as the other GET
```suggestion
This method does not support OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).
```

## Request headers

| Name          | Description    |
| :------------ | :------------- |
| Authorization | Bearer {token} |

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `200 OK` response code and a [servicePrincipalSignInActivity](../resources/serviceprincipalsigninactivity.md) object in the response body.

## Examples

### Example 1: Get the last sign-in information for a service principal

The following example shows how to get the last sign-in information for a service principal based on its ID.

#### Request

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_signin_1",
  "sampleKeys": ["ODNmNDUyOTYtZmI4Zi00YWFhLWEzOTktYWM1MTA4NGUwMmI3"]
}-->
```http
GET https://graph.microsoft.com/beta/reports/servicePrincipalSignInActivities/ODNmNDUyOTYtZmI4Zi00YWFhLWEzOTktYWM1MTA4NGUwMmI3
```

#### Response

The following is an example of the response.

<!-- {
  "blockType": "response",
  "truncated": true,

>   "@odata.type": "microsoft.graph.signIn"

Reviewer 2: Odata.type as signIn? Should we use servicePrincipalSignInActivity instead?

Author: updated.

     },
     "delegatedClientSignInActivity": {
          "lastSignInDateTime": "2021-01-01T00:00:00Z",
          "lastSignInRequestId": "2d245633-0f48-4b0e-8c04-546c2bcd61f5"
     },
     "delegatedResourceSignInActivity": {
          "lastSignInDateTime": "2021-02-01T00:00:00Z",
          "lastSignInRequestId": "d2b4c623-f930-42b5-9519-7851ca604b16"
     },
     "id": "ODNmNDUyOTYtZmI4Zi00YWFhLWEzOTktYWM1MTA4NGUwMmI3",
     "lastSignInActivity": {
          "lastSignInDateTime": "2021-04-01T00:00:00Z",
          "lastSignInRequestId": "cd9733e8-d75a-468f-a63d-6e82bd48c05e"
     }
}
```

### Example 2: Get a servicePrincipalSignInActivity filtered by applicationId

The following example shows how to get a **servicePrincipalSignInActivity** object filtered by **applicationId**.

#### Request

The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_serviceprincipal_last_signin_3"
}-->

```http
GET https://graph.microsoft.com/beta/reports/servicePrincipalSignInActivities?$filter=appId eq 'f4d9654f-0305-4072-878c-8bf266dfe146'
```

#### Response

The following is an example of the response.

<!-- {
  "blockType": "response",
  "truncated": true,

>   "@odata.type": "microsoft.graph.signIn"

Reviewer 2: Odata.type as signIn? Should we use servicePrincipalSignInActivity instead?
